tests/std/include/range_algorithm_support.hpp

```diff
@@ -247,6 +249,35 @@ namespace test {
         }
     };
 
+    template <class Ref>
+    struct common_reference {
+        Ref ref_;
+
+        common_reference(Ref ref) : ref_{static_cast<Ref>(ref)} {}

```

Comment:
Naming nitpick: I know it's test code, and it should be fairly hard for this header to be included after a `using namespace std;`, but `ref` is the name of a Standard function, and I usually recommend avoiding such identifiers in test code. (It makes global searching easier, for one thing.)